src/components/Dropdown/Dropdown.test.tsx

```diff
@@ -20,77 +19,106 @@ const DropdownTest = (props: Partial<DropdownProps> = {}) => {
-  render(<DropdownTest open={true} aria-label='test' style={{ color: 'red' }} />)
+it('should extend DropdownMenu props', async () => {
+  await act(async () => {
+    render(<DropdownTest open={true} aria-label='test' style={{ color: 'red' }} />)
+  })
   expect(document.body).toMatchSnapshot()
 })
 
-it('should control aria attributes on anchor element', () => {
-  const { getByText, rerender } = render(<DropdownTest open={false} />)
+it('should control aria attributes on anchor element', async () => {
+  let rerender: RenderResult['rerender']
+  let getByText: RenderResult['getByText']
+
+  await act(async () => {
+    const result = render(<DropdownTest open={false} />)
+    rerender = result.rerender
+    getByText = result.getByText
+  })
+
   const anchor = getByText('Anchor')
 
   expect(anchor.getAttribute('aria-haspopup')).toEqual('true')
   expect(anchor.getAttribute('aria-expanded')).toBeFalsy()
   expect(anchor.getAttribute('aria-controls')).toBeFalsy()
 
-  rerender(<DropdownTest open={true} />)
+  await act(async () => {
+    rerender(<DropdownTest open={true} />)
+  })
 
   expect(anchor.getAttribute('aria-haspopup')).toEqual('true')
   expect(anchor.getAttribute('aria-expanded')).toEqual('true')
   expect(anchor.getAttribute('aria-controls')).toEqual('dropdown-abc')
 })
 
-it('should call "onClose" when "Escape" is pressed and dropdown is open', () => {
+it('should call "onClose" when "Escape" is pressed and dropdown is open', async () => {
   const close = jest.fn()
-  const { rerender } = render(<DropdownTest open={false} onClose={close} />)
+
+  let rerender: RenderResult['rerender']
+  await act(async () => {
+    rerender = render(<DropdownTest open={false} onClose={close} />).rerender
+  })
 
   expect(close).not.toHaveBeenCalled()
 
   fireEvent.keyDown(document.body, { key: 'Escape' })
   expect(close).not.toHaveBeenCalled()
 
-  rerender(<DropdownTest open={true} onClose={close} />)
+  await act(async () => {
+    rerender(<DropdownTest open={true} onClose={close} />)
+  })
   fireEvent.keyDown(document.body, { key: 'Escape' })
   expect(close).toHaveBeenCalledTimes(1)
 
-  rerender(<DropdownTest open={false} onClose={close} />)
+  await act(async () => {
+    rerender(<DropdownTest open={false} onClose={close} />)
+  })
   fireEvent.keyDown(document.body, { key: 'Escape' })
   expect(close).toHaveBeenCalledTimes(1)
 })
 
 it('should call "onClose" when focus go outside menu', async () => {
   const close = jest.fn()
-  render(<DropdownTest open={true} onClose={close} />)
+  await act(async () => {
+    render(<DropdownTest open={true} onClose={close} />)
+  })
 
-  expect(document.activeElement).toEqual(document.body)
+  // expect(document.activeElement).toEqual(document.body)
 
   expect(close).not.toHaveBeenCalled()
 
   fireEvent.blur(document.body.querySelector('ul'))
-  await wait()
-  expect(close).toHaveBeenCalledTimes(1)
+  await waitFor(() => expect(close).toHaveBeenCalledTimes(1))
```

Comment:
This assertion is still failing, not sure why, `blur` fires correctly but `activeElement` stays an `li` from the `DropdownMenu`.

It works correctly on the storybook, tho.

Reply:
changed to `element.focus()` to replicate the focus/blur correctly.